crates/ty_python_semantic/src/types/builder.rs

```diff
@@ -202,7 +202,7 @@ enum ReduceResult<'db> {
 
 // TODO increase this once we extend `UnionElement` throughout all union/intersection
 // representations, so that we can make large unions of literals fast in all operations.
-const MAX_UNION_LITERALS: usize = 200;
+const MAX_UNION_LITERALS: usize = 100;
```

Comment:
This tweak is necessary to avoid crashes in cases like this:

```py
def f():
    x = 1
    def g():
        nonlocal x
        x += 1
```

Previously we would exceed the cycle limit before we gave up on the int literal union and called `x` an `int`. This was @carljm's suggested fix.

Reply:
(I filed https://github.com/astral-sh/ty/issues/957 to track this separately.)

Reply:
Would it make sense to make this change in a separate PR to understand its performance and ecosystem impact?